src/type.js

```diff
@@ -56,6 +56,24 @@ const validators = {
 	},
 
 	/**
+	 * Creates a validator that checks a value against a single type or null.
+	 * @param {function()} typeValidator Validator to check value against.
+	 * @return {function()} Validator.
+	 */
+	maybe: function(typeValidator) {
+		return (value, name, context) => {
+			const validation = typeValidator(value, name);
+
+			if (value === null || !(validation instanceof Error)) {
```

Comment:
I think it's worth also accepting `undefined` here like suggested in the [issue](https://github.com/metal/metal-state-validators/issues/7#issuecomment-227778770).

Other than that this lgtm :)